### PR TITLE
get-slide-metadata snippet for PowerPoint

### PR DESCRIPTION
**Pull request by Rick-Kirkham**
Uses the SlideRange coercion type

### CODE REVIEW

File: samples/powerpoint/basics/get-slide-metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+id: ''
+gist: ''
+source: POWERPOINT
+author: Microsoft
+name: get-slide-metadata
```
Comment (jakobpn) [inline samples/powerpoint/basics/get-slide-metadata.yaml]: get-side-metadata -> Get slide metadata

File: samples/powerpoint/basics/get-slide-metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+description: 'Gets the title, index, and ID of the selected slide(s).'
+script:
+  content: |-
+    $("#run").click(run);
+
+    function run() {
```
Comment (jakobpn) [inline samples/powerpoint/basics/get-slide-metadata.yaml]: run -> getSlideMetadata

File: samples/powerpoint/basics/get-slide-metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  content: "body {\r\n    margin: 0;\r\n    padding: 10px;\r\n}\r\n\r\n\r\n/* Button customization, including overwriting some Fabric defaults */\r\n\r\n.ms-Button, .ms-Button:focus {\r\n    background: #b7472a;\r\n    border: #b7472a;\r\n}\r\n\r\n    .ms-Button > .ms-Button-label,\r\n    .ms-Button:focus > .ms-Button-label,\r\n    .ms-Button:hover > .ms-Button-label {\r\n        color: white;\r\n    }\r\n\r\n    .ms-Button:hover, .ms-Button:active {\r\n        background: #8e3720;\r\n    }\r\n\r\n    .ms-Button.is-disabled, .ms-Button:disabled {\r\n        background-color: #f4f4f4;\r\n        border-color: #f4f4f4;\r\n    }\r\n\r\n    .ms-Button.is-disabled .ms-Button-label,\r\n    .ms-Button:disabled .ms-Button-label {\r\n        color: #a6a6a6;\r\n    }\r\n\r\n    "
+  language: css
+template:
+  content: >
+    <p class="ms-font-m">Select one or more slides and click <b>Run</b> to get
+    the ID, title, and index of the slide(s).</p>
+
+
+    <button id="run" class="ms-Button">
+        <span class="ms-Button-label">Run</span>
```
Comment (jakobpn) [inline samples/powerpoint/basics/get-slide-metadata.yaml]: run -> Get slide metadata

File: samples/powerpoint/basics/get-slide-metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            }
+        );
+    }
+  language: typescript
+style:
+  content: "body {\r\n    margin: 0;\r\n    padding: 10px;\r\n}\r\n\r\n\r\n/* Button customization, including overwriting some Fabric defaults */\r\n\r\n.ms-Button, .ms-Button:focus {\r\n    background: #b7472a;\r\n    border: #b7472a;\r\n}\r\n\r\n    .ms-Button > .ms-Button-label,\r\n    .ms-Button:focus > .ms-Button-label,\r\n    .ms-Button:hover > .ms-Button-label {\r\n        color: white;\r\n    }\r\n\r\n    .ms-Button:hover, .ms-Button:active {\r\n        background: #8e3720;\r\n    }\r\n\r\n    .ms-Button.is-disabled, .ms-Button:disabled {\r\n        background-color: #f4f4f4;\r\n        border-color: #f4f4f4;\r\n    }\r\n\r\n    .ms-Button.is-disabled .ms-Button-label,\r\n    .ms-Button:disabled .ms-Button-label {\r\n        color: #a6a6a6;\r\n    }\r\n\r\n    "
+  language: css
+template:
+  content: >
+    <p class="ms-font-m">Select one or more slides and click <b>Run</b> to get
```
Comment (jakobpn) [inline samples/powerpoint/basics/get-slide-metadata.yaml]: Change to "Demonstrates how to get slide metadata"

File: samples/powerpoint/basics/get-slide-metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  content: |-
+    $("#run").click(run);
+
+    function run() {
+        Office.context.document.getSelectedDataAsync(Office.CoercionType.SlideRange,
+            function (asyncResult) {
+                if (asyncResult.status === Office.AsyncResultStatus.Failed) {
+                    console.log(asyncResult.error.message);
+                } else {                
+                    document.getElementById('output').innerText = JSON.stringify(asyncResult.value);
```
Comment (jakobpn) [inline samples/powerpoint/basics/get-slide-metadata.yaml]: Output to console instead

Comment (jakobpn) [reply samples/powerpoint/basics/get-slide-metadata.yaml]: Use console.log(JSON.stringify(asyncResult.value, null, 4));

**Review by jakobpn — CHANGES_REQUESTED** [review]

File: samples/powerpoint/basics/get-slide-metadata.yaml (hunk 2 above)
Comment (Rick-Kirkham) [reply samples/powerpoint/basics/get-slide-metadata.yaml]: This conflicts with the guidance in the Readme for wrathofzombies/samples:
"Each button-click handler should have its own async function, called "run" if there is only one button on the page "

File: samples/powerpoint/basics/get-slide-metadata.yaml (hunk 4 above)
Comment (Rick-Kirkham) [reply samples/powerpoint/basics/get-slide-metadata.yaml]: Users need to be told to select one or more slides. Also, your wording just repeats what the title says, so it seems redundant.

File: samples/powerpoint/basics/get-slide-metadata.yaml (hunk 3 above)
Comment (Rick-Kirkham) [reply samples/powerpoint/basics/get-slide-metadata.yaml]: The pattern with other snippets is to label the button either "Run" or "Run Code".

File: samples/powerpoint/basics/get-slide-metadata.yaml (hunk 4 above)
Comment (Rick-Kirkham) [reply samples/powerpoint/basics/get-slide-metadata.yaml]: Added that, but kept the original paragraph too. It is needed.

File: samples/powerpoint/basics/get-slide-metadata.yaml (hunk 1 above)
Comment (Rick-Kirkham) [reply samples/powerpoint/basics/get-slide-metadata.yaml]: done

File: samples/powerpoint/basics/get-slide-metadata.yaml (hunk 5 above)
Comment (Rick-Kirkham) [reply samples/powerpoint/basics/get-slide-metadata.yaml]: done

File: samples/powerpoint/basics/get-slide-metadata.yaml (hunk 2 above)
Comment (jakobpn) [reply samples/powerpoint/basics/get-slide-metadata.yaml]: OK. I will change the readme file.

File: samples/powerpoint/basics/get-slide-metadata.yaml (hunk 3 above)
Comment (jakobpn) [reply samples/powerpoint/basics/get-slide-metadata.yaml]: Please see samples\excel\range\set-get-values.yaml for the pattern

**Review by jakobpn — CHANGES_REQUESTED** [review]

File: samples/powerpoint/basics/get-slide-metadata.yaml (hunk 2 above)
Comment (Rick-Kirkham) [reply samples/powerpoint/basics/get-slide-metadata.yaml]: ok

File: samples/powerpoint/basics/get-slide-metadata.yaml (hunk 3 above)
Comment (Rick-Kirkham) [reply samples/powerpoint/basics/get-slide-metadata.yaml]: ok

**Pull request merged**